### PR TITLE
Update app service and DC

### PR DESCRIPTION
**Pull request by pamelafox**
Getting https://github.com/Azure-Samples/azure-flask-postgres-flexible-appservice/ to deploy, and some DC updates.

### CODE REVIEW

File: templates/docker_compose_postgres.yml [hunk 1]
```diff
@@ -7,9 +7,9 @@ image: postgres:14
     environment:
       POSTGRES_USER: postgres
       POSTGRES_DB: relecloud
-      POSTGRES_HOST: db
       POSTGRES_PASSWORD: postgres
-      POSTGRES_PORT: 5432
```
Comment (pamelafox) [inline templates/docker_compose_postgres.yml:12]: The postgres image doesn't use _HOST or _PORT env variables, so these were having no effect

File: templates/docker_compose_postgres.yml [hunk 2]
```diff
@@ -26,6 +26,5 @@ environment:
       POSTGRES_USERNAME: postgres
       POSTGRES_DATABASE: relecloud
       POSTGRES_HOST: db
-      POSTGRES_PORT: 5432
```
Comment (pamelafox) [inline templates/docker_compose_postgres.yml:29]: Our code doesnt use POSTGRES_PORT so I removed it

File: templates/docker_compose_postgres.yml [hunk 3]
```diff
@@ -7,9 +7,9 @@ image: postgres:14
     environment:
       POSTGRES_USER: postgres
       POSTGRES_DB: relecloud
-      POSTGRES_HOST: db
       POSTGRES_PASSWORD: postgres
-      POSTGRES_PORT: 5432
+
+    restart: unless-stopped
```
Comment (pamelafox) [inline templates/docker_compose_postgres.yml:12]: Added per #158

File: {{cookiecutter.__src_folder_name}}/infra/main.bicep [hunk 4]
```diff
@@ -131,8 +131,10 @@ module web 'web.bicep' = {
     dbserverDomainName: db.outputs.dbserverDomainName
     dbserverUser: db.outputs.dbserverUser
     dbserverDatabaseName: db.outputs.dbserverDatabaseName
+    {% if cookiecutter.project_host == "aca" %}
```
Comment (pamelafox) [inline {{cookiecutter.__src_folder_name}}/infra/main.bicep:134]: For app service, we retrieve password via KeyVault ref

File: {{cookiecutter.__src_folder_name}}/.devcontainer/devcontainer.json [hunk 5]
```diff
@@ -25,13 +25,14 @@
                 "charliermarsh.ruff",
                 "ms-python.python",
                 "ms-python.black-formatter",
-                "microsoft.bicep",
```
Comment (pamelafox) [inline {{cookiecutter.__src_folder_name}}/.devcontainer/devcontainer.json:28]: This extension does not seem to exist

File: {{cookiecutter.__src_folder_name}}/.devcontainer/devcontainer.json [hunk 6]
```diff
@@ -25,13 +25,14 @@
...
-                "bierner.github-markdown-preview",
+                {% if cookiecutter.project_host == "aca" %}
+                "ms-azuretools.vscode-docker",
```
Comment (pamelafox) [inline {{cookiecutter.__src_folder_name}}/.devcontainer/devcontainer.json:29]: This one should be conditional on ACA. Moved it up for comma reasons

**Comment by pamelafox** [comment]
Updated to add a launch.json for Flask

File: {{cookiecutter.__src_folder_name}}/.devcontainer/devcontainer.json (hunk 5 above)
Comment (kjaymiller) [reply {{cookiecutter.__src_folder_name}}/.devcontainer/devcontainer.json:28]: The extension id is - `ms-azuretools.vscode-bicep`. Can we add that in?

**Review by kjaymiller — APPROVED** [review]
Thanks for adding all these changes! Looks Good. Please see one comment but that can be added in a separate PR is necessary.

**Comment by kjaymiller** [comment]
> Thanks for adding all these changes! Looks Good. Please see one comment but that can be added in a separate PR is necessary.

Issue for the one comment was created so feel free to merge

**Pull request merged**